dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/acl/DefaultTrackerAccessManager.java

```diff
@@ -108,26 +127,22 @@ private List<String> canRead(
                     p.isRegistration()
                         && Objects.equals(
                             p.getTrackedEntityType().getUid(),
-                            trackedEntity.getTrackedEntityType().getUid()))
+                            trackedEntity.getTrackedEntityType().getUid())
+                        && aclService.canDataRead(user, p))
```

Comment:
Will `programService.getAllPrograms()` return all programs the user has `metadata` read access to?

Reply:
yes